src/sort/mod.rs

```diff
@@ -145,3 +146,29 @@ impl<T: IntoSort> IntoSort for Option<T> {
 
 pub type PreSort =
     fn(typeinfo: &mut TypeInfo, name: Symbol, params: &[Expr]) -> Result<ArcSort, TypeError>;
+
+pub(crate) struct ValueEq {
+    pub unit: Arc<UnitSort>,
+}
+
+impl PrimitiveLike for ValueEq {
+    fn name(&self) -> Symbol {
```

Comment:
optional: 

A while back, I noticed some performance problems related to calling "string constant".into() all over the place. It was faster / cheaper to have a lazy_static! that cached the intern value and then loading that here.

Reply:
Done!